### PR TITLE
ACS-1582 : Storage classes - java api

### PR DESCRIPTION
**Pull request by tzclucian**

### CODE REVIEW

File: repository/src/main/java/org/alfresco/service/cmr/repository/ContentService.java [hunk 1]
```diff
@@ -170,4 +173,72 @@ public ContentWriter getWriter(NodeRef nodeRef, QName propertyQName, boolean upd
      */
     @Auditable(parameters = {"nodeRef", "expiresAt"})
     public DirectAccessUrl getDirectAccessUrl(NodeRef nodeRef, Date expiresAt);
+
+    /**
+     * Checks if the store supports the retrieving of direct access URLs.
+     *
+     * @return true if direct access URLs retrieving is supported, false otherwise
+     */
+    default boolean isDirectAccessSupported()
```
Comment (ancutam) [inline repository/src/main/java/org/alfresco/service/cmr/repository/ContentService.java:182]: I think this one is part of the other feature :)

Comment (tzclucian) [reply repository/src/main/java/org/alfresco/service/cmr/repository/ContentService.java]: Thank you, updated :)

File: repository/src/main/java/org/alfresco/repo/content/AbstractRoutingContentStore.java [hunk 2]
```diff
@@ -416,4 +417,11 @@ public boolean delete(String contentUrl) throws ContentIOException
         }
         return deleted;
     }
+
+    @Override
+    public boolean isStorageClassesSupported(Set<String> storageClasses)
+    {
+        // TODO: what should we do here? check whether at least one supports it similar to isContentUrlSupported?
```
Comment (cristianturlica) [inline repository/src/main/java/org/alfresco/repo/content/AbstractRoutingContentStore.java:424]: Yes, it should return true if at least one store supports it, similar to `isContentUrlSupported`.

File: repository/src/main/java/org/alfresco/repo/content/filestore/FileContentStore.java [hunk 3]
```diff
@@ -639,4 +640,10 @@ private void ensureFileInContentStore(File file)
             throw new ContentIOException("Access to files outside of content store root is not allowed: " + file);
         }
     }
+
+    @Override
+    public boolean isStorageClassesSupported(Set<String> storageClasses)
```
Comment (cristianturlica) [inline repository/src/main/java/org/alfresco/repo/content/filestore/FileContentStore.java:645]: We could consider having this at a level closer to the ContentStore hierarchy.

File: repository/src/main/java/org/alfresco/service/cmr/repository/ContentService.java [hunk 4]
```diff
@@ -170,4 +173,62 @@ public ContentWriter getWriter(NodeRef nodeRef, QName propertyQName, boolean upd
...
     @Auditable(parameters = {"nodeRef", "expiresAt"})
     public DirectAccessUrl getDirectAccessUrl(NodeRef nodeRef, Date expiresAt);
+
+    /**
+     * Checks whether or not the current {@link ContentService} supports the provided {@link Set} storage classes
+     *
+     * @param storageClasses The storage classes that will be checked whether or not are supported
+     * @return true if the storage classes are supported, false otherwise.
+     */
+    default boolean isStorageClassesSupported(Set<String> storageClasses)
```
Comment (cristianturlica) [inline repository/src/main/java/org/alfresco/service/cmr/repository/ContentService.java:183]: Default storage class (e.g. null) should always be supported.

File: repository/src/main/java/org/alfresco/service/cmr/repository/ContentService.java [hunk 5]
```diff
@@ -170,4 +173,62 @@ public ContentWriter getWriter(NodeRef nodeRef, QName propertyQName, boolean upd
...
+    /**
+     * Updates the storage class for a {@link NodeRef}
+     *
+     * @param nodeRef The ref of the node that will have its storage classes updated
+     * @param storageClasses The new storage classes
+     * @param parameters extra parameters
+     */
+    default void updateStorageClasses(NodeRef nodeRef, Set<String> storageClasses, Map<String, Object> parameters)
+    {
+
```
Comment (cristianturlica) [inline repository/src/main/java/org/alfresco/service/cmr/repository/ContentService.java:205]: We should throw an exception if this isn't implemented and someone tries to use it.

Comment (tzclucian) [reply repository/src/main/java/org/alfresco/service/cmr/repository/ContentService.java:205]: I haven't got there with the code yet but noted.

File: repository/src/main/java/org/alfresco/repo/content/filestore/FileContentStore.java (hunk 3 above)
Comment (tzclucian) [reply repository/src/main/java/org/alfresco/repo/content/filestore/FileContentStore.java]: Moved the code snippet, as discussed regarding `default` to the `ContentStore`

File: repository/src/test/java/org/alfresco/repo/content/filestore/FileContentStoreTest.java [hunk 6]
```diff
@@ -347,6 +348,18 @@ public void testSpoofedContent() throws Exception
...
+    @Test
+    public void testSupportsDefaultStorageClass()
+    {
+        assertTrue(store.isStorageClassesSupported(Set.of(ContentStore.DEFAULT_SC)));
+    }
+
+    @Test
+    public void testDoesNotSupportUnknownStorageClass()
+    {
+        assertTrue(store.isStorageClassesSupported(Set.of("unknown")));
```
Comment (DenisGabriela) [inline repository/src/test/java/org/alfresco/repo/content/filestore/FileContentStoreTest.java:361]: Does this test pass?

Comment (tzclucian) [reply repository/src/test/java/org/alfresco/repo/content/filestore/FileContentStoreTest.java]: Fixed, thank you.

**Pull request merged**